### ISSUE TITLE
Istio Ingress gateway ends up broken when SDS fails

### ISSUE THREAD

**Issue by fbsb** [istio/istio#31245]
**Bug description**

Sometimes an ingress gateway pod fails while getting SDS secrets and ends up broken responding with 503 to all requests.

This seems to happen randomly so it is not easily reproducible.

After finding a broken pod and enabling debug logs this pops up:

```
2021-03-04T16:54:55.043313Z	debug	envoy wasm	expr_evaluate value error: No value with name "upstream_peer" found in Activation
2021-03-04T16:54:55.043326Z	debug	envoy wasm	wasm log: [extensions/stats/plugin.cc:609]::report() metricKey cache hit , stat=12
2021-03-04T16:54:55.043330Z	debug	envoy wasm	wasm log: [extensions/stats/plugin.cc:609]::report() metricKey cache hit , stat=6
2021-03-04T16:54:55.043332Z	debug	envoy wasm	wasm log: [extensions/stats/plugin.cc:609]::report() metricKey cache hit , stat=10
2021-03-04T16:54:55.043335Z	debug	envoy wasm	wasm log: [extensions/stats/plugin.cc:609]::report() metricKey cache hit , stat=14
2021-03-04T16:54:55.046567Z	debug	envoy router	[C15292569][S3750510621003656557] performing retry
2021-03-04T16:54:55.046587Z	debug	envoy pool	queueing request due to no available connections
2021-03-04T16:54:55.046590Z	debug	envoy pool	creating a new connection
2021-03-04T16:54:55.046594Z	debug	envoy config	Create NotReadySslSocket
2021-03-04T16:54:55.046615Z	debug	envoy client	[C15292848] connecting
2021-03-04T16:54:55.046619Z	debug	envoy connection	[C15292848] connecting to 100.70.211.182:8000
2021-03-04T16:54:55.046706Z	debug	envoy connection	[C15292848] connection in progress
2021-03-04T16:54:55.046915Z	debug	envoy connection	[C15292848] connected
2021-03-04T16:54:55.046919Z	debug	envoy connection	[C15292848] closing socket: 0
2021-03-04T16:54:55.046942Z	debug	envoy client	[C15292848] disconnect. resetting 0 pending requests
2021-03-04T16:54:55.046946Z	debug	envoy pool	[C15292848] client disconnected, failure reason: TLS error: Secret is not supplied by SDS
2021-03-04T16:54:55.046952Z	debug	envoy router	[C15292569][S3750510621003656557] upstream reset: reset reason connection failure
2021-03-04T16:54:55.046968Z	debug	envoy http	[C15292569][S3750510621003656557] Sending local reply with details upstream_reset_before_response_started{connection failure,TLS error: Secret is not supplied by SDS}
2021-03-04T16:54:55.046988Z	debug	envoy http	[C15292569][S3750510621003656557] encoding headers via codec (end_stream=false):
':status', '503'
'content-length', '91'
'content-type', 'text/plain'
'date', 'Thu, 04 Mar 2021 16:54:54 GMT'
'server', 'istio-envoy'
```

We noticed issues appearing in relatively small clusters but with high service and pod density during rolling redeployments.

Also we did not see this issue with normal workload pods.

This issue seems related to https://github.com/istio/istio/issues/22443

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

I expected the pod to retry fetching SDS secrets after a failure or get not ready.

**Steps to reproduce the bug**

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

1.7.7

**How was Istio installed?**

**Environment where the bug was observed (cloud vendor, OS, etc)**

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by howardjohn** [comment]
When you do see this, if you do `istioctl pc secret` do you see ROOTCA warming? If so, its probably the same as https://github.com/istio/istio/issues/22443 which is fixed in 1.8

**Comment by howardjohn** [comment]
This is a known issue in 1.7 which is no longer supported https://istio.io/latest/news/support/announcing-1.7-eol-final/. Please upgrade! Thanks
